go.mod

```diff
@@ -1,13 +1,14 @@
 module github.com/ceph/go-ceph
 
 go 1.21
+toolchain go1.22.5
```

Comment:
I believe this addition is due to the updated minimum Go version for the package which is greater than the version specified just above in the "go" line.
> <h2> Module Highlights </h2>
>
> - github.com/aws/smithy-go: v1.22.3
> - Dependency Update: Bump minimum Go version to 1.22 per our language support policy.

@phlogistonjohn Does that mean we bump our Go version too?

Reply:
I had to (re)read https://go.dev/blog/toolchain to remind myself what this is. I think it'd be OK to bump up the `go` line to `1.22` and set `toolchain` to match. I do not think we need a 'z' value unless Go objects.

I think we should tackle this is a separate PR and then rebase this and other dependency PRs afterwards so that they only change the dependencies and not other properties of the go.mod file.

Reply:
I've verified that with https://github.com/ceph/go-ceph/pull/1085 `go mod tidy` no longer adds the `toolchain` line to _go.mod_.